backend/app/datasets/service.py

```diff
@@ -74,6 +75,27 @@ def __init__(self, db: Session):
         self.db = db
         self.namespace_validator = NamespaceValidator(DatasetModel)
 
+    @staticmethod
+    def _build_prefix_tsquery(query: str) -> str | None:
+        """
+        Build a prefix tsquery string like 'foo:* & bar:*' from free text.
+
+        - Tokenizes on non-word characters.
+        - Filters out tokens shorter than 2 characters (e.g., single-character tokens are dropped).
+        - Appends ':*' to each token to enable prefix matching.
+        - Returns None if no valid tokens remain after filtering.
+
+        Example:
+            'Hello world!' becomes 'hello:* & world:*'
+            'a b' returns None (all tokens too short)
+        """
+        tokens = [t for t in re.split(r"\W+", (query or "").lower()) if len(t) >= 2]
+        if not tokens:
+            return None
+        prefixed = [f"{t}:*" for t in tokens]
```

Comment:
Now we match `app` to `apple`, is it also feasible to place a start at the front of the word? So we can match `pple` to `apple`?